script/search/2.0/search.inc.php

```diff
@@ -40,31 +58,21 @@ function GetSearchMatches($platform, $query, $pageNumber) {
         $result->platform = 'web';
       }
 
-      $result->filter = KeyboardSearchResult::FILTER_SEARCH;
-      $result->regionmatch = false;
-
-      $textparts = explode(':', $query);
+      $query = trim($query);
 
-      for($i = 0; $i < sizeof($textparts) - 1; $i++) {
-        $match = strtolower($textparts[$i]);
-        if(!strcmp($match, 'bcp47')) $result->filter = KeyboardSearchResult::FILTER_LANGUAGE;
-        else if(!strcmp($match, 'id')) $result->filter = KeyboardSearchResult::FILTER_ID;
-        else if(!strcmp($match, 'all')) $result->allmatch = true;
-        else if(!strcmp($match, 'region')) $result->regionmatch = true;
-        else if(!strcmp($match, 'legacy')) $result->filter = KeyboardSearchResult::FILTER_LEGACY;
+      if(preg_match('/^('.implode('|',array_keys(KeyboardSearch::FILTERS)).')(.+)$/', $query, $matches)) {
+        $result->text = $matches[2];
+        $result->filter = KeyboardSearch::FILTERS[$matches[1]];
```

Comment:
So, just one filter at a time, then?

Reply:
Yes; there aren't really any valuable mix and match filters for this search that I could come up with -- one is sufficient.